CSNotificationView/CSNotificationView.m

```diff
@@ -383,6 +383,10 @@ - (void)dismissWithStyle:(CSNotificationViewStyle)style message:(NSString *)mess
 - (CGFloat)topLayoutGuideLengthCalculation
 {
     CGFloat top = MIN([UIApplication sharedApplication].statusBarFrame.size.height, [UIApplication sharedApplication].statusBarFrame.size.width);
+
+    UIView *view = self.parentNavigationController.view ?: self.parentViewController.view;
+    CGPoint origin = [view.superview convertPoint:view.frame.origin toView:[UIApplication sharedApplication].keyWindow.rootViewController.view];
```

Comment:
Why are you deriving the origin from the application's root view controller? This may fit your special usecase but it is not a general-purpose solution.